### DISCUSSION TITLE
NetBox Plugin migration files diff between v3.4 with less v3.4

### DISCUSSION THREAD

**Discussion by miyuk** [netbox-community/netbox#11889] [Q&A] [answered]
Now, I develop netbox plugin in NetBox v3.4.4.

I use it in NetBox v3.3.x and execute `python manage.py migrate`.

However I got a error with `utilities.json is not found` at below line.
```
import utilities.json
```

This module is added in v3.4 for Decimal field.

https://github.com/netbox-community/netbox/issues/10348

This PR is good, however, it is not good to be spoiled compatibility.

Do you have an ides to resolve this.

or Should we develop in NetBox v3.3.x?

**Comment by candlerb** [comment, marked as answer]
> I develop netbox plugin in NetBox v3.4.4.
> 
> I use it in NetBox v3.3.x

Your conclusion is correct: if you want your plugin to work in v3.3, it's safest to develop it with v3.3.  If you want it to work with v3.4, then develop with v3.4.  If you want to it to work with both v3.3 and v3.4, then you may have to add extra code to handle the differences.  There is an example [here](https://github.com/netbox-community/reports/blob/master/scripts/multi_connect.py#L17-L25).

`utilities.json` is an internal part of Netbox, and if you choose to use them there are no stability guarantees. For that matter, even core models and the public REST API can change between versions in incompatible ways, and often do.  This is something that you need to track as a plugin writer, script writer, API consumer etc.  The release notes call out the most important changes.

**Reply by miyuk** [reply]
thank you for your reply.

I understood that.

I will make extra code to handle the differences sometimes.
